Mathlib/Algebra/Ring/Divisibility/Lemmas.lean

```diff
@@ -117,3 +117,30 @@ lemma dvd_mul_sub_mul_mul_gcd_of_dvd {p a b c d x y : R} [IsDomain R] [GCDMonoid
     dvd_mul_sub_mul_mul_right_of_dvd h1 h2⟩
 
 end CommRing
+
+section misc
+
+variable [Ring R] [LinearOrder R] {x y : R}
+
+theorem Associated.abs_left (h : Associated x y) :
+    Associated |x| y := by
+  obtain ha | ha := abs_choice x
+  · simpa [ha]
+  · simpa [ha] using h.neg_left
+
+theorem Associated.abs_right (h : Associated x y) :
+    Associated x |y| :=
+  (h.symm.abs_left).symm
```

Comment:
I guess you could save yourself a few lines here with
```
alias ⟨_, Associated.abs_left⟩ := associated_abs_left_iff
alias ⟨_, Associated.abs_right⟩ := associated_abs_right_iff
```
since you didn't use these to prove the `iff`.

Reply:
Yes, that's a nice trick